projects/chaos/src/sequencer_fsm.c

```diff
@@ -155,6 +164,17 @@ FSM_STATE_TRANSITION(sequencer_state_drive) {
   FSM_ADD_TRANSITION(CHAOS_EVENT_SEQUENCE_RESET, sequencer_state_drive);
 }
 
+static bool prv_is_valid_transition(ChaosEventSequence present, ChaosEventSequence next) {
```

Comment:
Shouldn't this already be part of your FSM?

Reply:
Yes but I can't check if a transition will succeed prior to attempting the transition which isn't what I want to do. I just want to check if I can transition ahead of time. Unless I missed something in the FSM API that allows for this check?

This is necessary in order to let driver controls know if an event will transition successfully. I know it is a redundancy but I am unaware of another way to "Peek" to see if the transition will work without re-encoding that information in a guard.

Reply:
Hmm. We could modify the transition tables to return true/false and the new state. `fsm_process_event` could use that and actually set the state/call the output function.